new_smart_launch.sh

```diff
@@ -34,10 +60,23 @@ fi
 # Setting allow-commands option
 factorio_command="$factorio_command --allow-commands $FACTORIO_ALLOW_COMMANDS"
 # Setting auto-pause option
-if [ "$FACTORIO_NO_AUTO_PAUSE" == true ] 
+if [ "$FACTORIO_NO_AUTO_PAUSE" == true ]
 then
 factorio_command="$factorio_command --no-auto-pause"
 fi
+# Include server-settings.json if one or more variables are populated
+if [ "$FACTORIO_SERVER_NAME" != "factorio server" ] \
+|| [ -n "$FACTORIO_SERVER_DESCRIPTION" ] \
```

Comment:
`-n` isn't exactly required in these tests according to http://stackoverflow.com/questions/3869072/test-for-non-zero-length-string-in-bash-n-var-or-var